src/logging/logging.bif

```diff
@@ -70,3 +70,87 @@ function Log::__flush%(id: Log::ID%): bool
 	bool result = zeek::log_mgr->Flush(id->AsEnumVal());
 	return zeek::val_mgr->Bool(result);
 	%}
+
+%%{
+namespace
+	{
+	zeek::EnumValPtr enum_ref(zeek::Val *v)
```

Comment:
I was surprised here that you needed to implement these, but I can't immediately find anything that would do this — it's sort of a `safe_make_intrusive()`... might be worth checking with @timwoj .

Reply:
Hmm, the `enum_ref` type check isn't absolutely required because static type checking of `Log::delay()` and `Log::__delay()` should capture that, but converting to an IntrusivePtr is still rather verbose.

On the Expr side of things we have helpers (though they abort on error):

I do find that conversions from `Val*` or `ValPtr` to a concrete/expected type pretty cumbersome to do safely, but also seems we may not overly worry ;-)
```
root@4ba3046bb535:/# zeek -e 'Log::write(Conn::LOG, "boom")'
fatal error in <command line>, line 1: Val::CONVERTER (string/record) (boom)
Aborted (core dumped)
```
---

On the Expr side of things we have helpers:

https://github.com/zeek/zeek/blob/a59578e018bc6a55b3d0eefffc76f07899800d80/src/Expr.cc#L134-L137


There is `Val::As()`, but that seems very dangerous if you're not sure that it's actually the right type and it is raw pointer based.

Something like `Val::TryAsPtr<T>` or `Val::AsPtr<T>` which return IntrusivePtrs and the former nil if the type doesn't match, the latter causing a fatal error. Both using using `NewRef{}` by default with maybe an extra template parameter to switch to `AdoptRef`, maybe?

Do we have something like this and I'm just not seeing it, or would that be worth having?